### ISSUE TITLE
Before moving to function definition, push the mark?

### ISSUE THREAD

**Issue by mitchellroe** [jacktasia/dumb-jump#311]
I've noticed that when using `dumb-jump-go`, I can't just pop the mark and get back to where I called `dumb-jump-go`.  I think it would be really useful to add the current location to the mark stack before moving to the function definition so one can easily get back to their last position with `C-u C-SPC` (or `set-mark-command` with the prefix argument).

**Comment by jacktasia** [comment]
Thanks for opening this. It's already pushing to the xref marker stack and there's already `dumb-jump-back` which is really just doing `xref-pop-marker-stack`. My understanding is this is the "proper" way to do it, but I could certainly be wrong. I am happy to consider other suggestions (or PRs) but I am not sure I follow exactly what you want instead.

**Comment by mitchellroe** [comment]
Ah, thank you.  I don't know how I missed `dumb-jump-back`.  I guess the muscle memory of C-u C-SPC is so ingrained I just figured that if it didn't work then something was wrong.  My mistake.

I've learned a little something about the global mark ring vs. the local mark ring, C-u C-SPC vs. C-x C-SPC, etc.  Without knowing more about how all these pieces fit together, I wouldn't feel comfortable making a recommendation for changing the way dumb-jump operates.  Sounds like it works just fine, just no the way I expected.  For the time being, I'll just train my fingers to use C-M-p when using dumb-jump and keep Reading the Manuals (TM).  Haha.  Thanks for your time.
